packages/styled-components/src/constructors/css.ts

```diff
@@ -5,14 +5,28 @@ import interleave from '../utils/interleave';
 import isFunction from '../utils/isFunction';
 import isPlainObject from '../utils/isPlainObject';
 
+/**
+ * Used when flattening object styles to determine if we should
+ * expand an array of styles.
+ */
+const addTag = (arg: ReturnType<typeof flatten> & { isCss?: boolean }) => {
+  if (Array.isArray(arg)) {
+    // eslint-disable-next-line no-param-reassign
+    arg.isCss = true;
+  }
+  return arg;
```

Comment:
Is this mutation required? or is it possible to:

```suggestion
  return Array.isArray(arg) ? { ...arg, isCss: true} : arg;
```

Reply:
In this case I think the OP's code is probably a little more performant since it isn't introducing additional objects into a hot path